tests/php/test-amp-options-rest-controller.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * Tests for AMP_Options_REST_Controller.
+ *
+ * @group options
+ *
+ * @package AMP
+ */
+
+use AmpProject\AmpWP\Option;
+
+/**
+ * Tests for AMP_Options_REST_Controller.
+ *
+ * @group amp-options
+ *
+ * @covers AMP_Options_REST_Controller
+ */
+class Test_AMP_Options_REST_Controller extends WP_UnitTestCase {
+
+	/**
+	 * Test instance.
+	 *
+	 * @var AMP_Options_REST_Controller
+	 */
+	private $controller;
+
+	/**
+	 * Set up.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		AMP_Options_Manager::update_option( Option::THEME_SUPPORT, AMP_Theme_Support::READER_MODE_SLUG );
+
+		do_action( 'rest_api_init' );
+		$this->controller = new AMP_Options_REST_Controller();
+	}
+
+	/**
+	 * Provides a test request.
+	 *
+	 * @return WP_REST_Request
+	 */
+	public function get_request() {
```

Comment:
I'm not sure this helper method is making the tests clearer.

It is only used twice, and the second time it is coupled with another direct instantiation of `WP_REST_Request` because a `POST` request is needed as well.